server/sonar-server/src/main/java/org/sonar/server/db/migrations/v453/AddCharacteristicUsabilityAndSubCharacteristicsComplianceMigration.java

```diff
@@ -0,0 +1,312 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.db.migrations.v453;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.utils.System2;
+import org.sonar.core.persistence.Database;
+import org.sonar.server.db.migrations.BaseDataChange;
+import org.sonar.server.db.migrations.Select;
+import org.sonar.server.db.migrations.Upsert;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+import java.sql.SQLException;
+import java.util.Date;
+import java.util.List;
+
+import static com.google.common.collect.Lists.newArrayList;
+
+/**
+ * See http://jira.codehaus.org/browse/SONAR-6187
+ *
+ * Add a new Characteristic 'Usability' with 2 sub-characteristics 'Accessibility' and 'Ease of Use'
+ * and add a new sub-characteristic 'Compliance' for all characteristics.
+ *
+ * @since 4.5.3
+ */
+public class AddCharacteristicUsabilityAndSubCharacteristicsComplianceMigration extends BaseDataChange {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AddCharacteristicUsabilityAndSubCharacteristicsComplianceMigration.class);
+
+  private static final String COMPLIANCE_NAME = "Compliance";
+
+  private final System2 system;
+
+  public AddCharacteristicUsabilityAndSubCharacteristicsComplianceMigration(Database db, System2 system) {
+    super(db);
+    this.system = system;
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    CharacteristicsContext characteristicsContext = new CharacteristicsContext(context, system);
+
+    int usabilityOder = moveCharacteristicsDownToBeAbleToInsertUsability(characteristicsContext);
+    createOrUpdateUsabilityCharacteristicAndItsSubCharacteristic(characteristicsContext, usabilityOder);
+
+    createSubCharacteristic(characteristicsContext, "REUSABILITY_COMPLIANCE", "Reusability " + COMPLIANCE_NAME, "REUSABILITY");
+    createSubCharacteristic(characteristicsContext, "PORTABILITY_COMPLIANCE", "Portability " + COMPLIANCE_NAME, "PORTABILITY");
+    createSubCharacteristic(characteristicsContext, "MAINTAINABILITY_COMPLIANCE", "Maintainability " + COMPLIANCE_NAME, "MAINTAINABILITY");
+    createSubCharacteristic(characteristicsContext, "SECURITY_COMPLIANCE", "Security " + COMPLIANCE_NAME, "SECURITY");
+    createSubCharacteristic(characteristicsContext, "EFFICIENCY_COMPLIANCE", "Efficiency " + COMPLIANCE_NAME, "EFFICIENCY");
+    createSubCharacteristic(characteristicsContext, "CHANGEABILITY_COMPLIANCE", "Changeability " + COMPLIANCE_NAME, "CHANGEABILITY");
+    createSubCharacteristic(characteristicsContext, "RELIABILITY_COMPLIANCE", "Reliability " + COMPLIANCE_NAME, "RELIABILITY");
+    createSubCharacteristic(characteristicsContext, "TESTABILITY_COMPLIANCE", "Testability " + COMPLIANCE_NAME, "TESTABILITY");
+  }
```

Comment:
Referring to comment above - isn't it a repeat :repeat: of string "_COMPLIANCE" :) ?